app.rb

```diff
@@ -0,0 +1,256 @@
+require_relative './book'
+require_relative './person'
+require_relative './rental'
+require_relative './teacher'
+require_relative './student'
+require_relative './classroom'
+
+def populate
+  grade9 = Classroom.new('Grade 9')
+  student1 = Student.new(17, grade9, 'std1', parent_permission: true)
+  student2 = Student.new(18, grade9, 'std2')
+  @students << (student1)
+  @students << (student2)
+  teacher = Teacher.new(29, 'Computer Science', 'Ms. Langtree')
+  teacher2 = Teacher.new(29, 'Computer Science', 'Ms. Lemon')
+  @teachers << (teacher)
+  @teachers << (teacher2)
+  book1 = Book.new('The Lord of the Rings', 'J. R. R. Tolkien')
+  book2 = Book.new('The Hobbit', 'J. R. R. Tolkien')
+  @books << book1
+  @books << book2
+end
+
+def display_menu
+  puts 'Welcome to School Library App!'.center(40, '-')
+  puts "\n"
+  puts '1 - List all books'
+  puts '2 - List all people'
+  puts '3 - Add a person'
+  puts '4 - Add a book'
+  puts '5 - Create a rental'
+  puts '6 - List all rentals for a given id.'
+  puts '7 - Exit'
+  print 'Enter your choice: '
+end
+
+def list_books
+  system 'cls'
+  puts 'List of books'.center(32)
+  count = 0
+  @books.each do |book|
+    puts "Book #{count += 1} Title: #{book.title} Author: #{book.author}"
+  end
+  puts ''
+  print 'press r to return to the main menu... : '
+  choice2 = gets.chomp
+  system 'cls' if %w[r R].include?(choice2)
+end
+
+def list_people
+  system 'cls'
+  puts 'List of people'.center(32)
+  count = 0
+  @students.each do |student|
+    puts "Student #{count += 1} Name: #{student.name} Age: #{student.age} "
+  end
+  count = 0
+  @teachers.each do |teacher|
+    puts "Teacher #{count += 1} Name: #{teacher.name} Age: #{teacher.age} "
+  end
+  puts ''
+  print 'press r to return to the main menu... : '
+  choice2 = gets.chomp
+  system 'cls' if %w[r R].include?(choice2)
+end
+
+def list_person
+  count = 0
+  @students.each do |student|
+    puts "Student #{count += 1} Name: #{student.name} ID: #{student.id} "
+  end
+  count = 0
+  @teachers.each do |teacher|
+    puts "Teacher #{count += 1} Name: #{teacher.name} ID: #{teacher.id} "
+  end
+  puts ''
+end
+
+def add_person
+  system 'cls'
+  loop do
+    print 'Do you want to add a student (1) or a teacher (2)? (Enter 3 to go back): '.center(10)
+    choice3 = gets.chomp.to_i
+    case choice3
+    when 1
+      add_student_loop
+    when 2
+      add_teacher_loop
+    when 3
+      system 'cls'
+      break
+    else
+      puts 'Invalid choice!'
+    end
+  end
+end
+
+def add_student_loop
+  loop do
+    add_student
+    system 'cls'
+    puts 'Student added!'
+    puts '1: Add another student'
+    puts '2: Return to previous menu'
```

Comment:
### Optional
- [ ] It will be nice if we fix the bug in the code, here is a screenshot to show what actually happens when we type the number to return to the main menu
![Screenshot from 2023-05-26 18-36-14](https://github.com/fuadmoin/school-library/assets/46388113/3869ed47-67f8-444e-b3a8-8f4b8d992e78)
![Screenshot from 2023-05-26 18-34-44](https://github.com/fuadmoin/school-library/assets/46388113/4b10486f-cf8c-4a12-8d44-100e5831600e)